[Bug Report] Wrong leading dimensions when multiplying FactoredMatrix by vector

If you are submitting a bug report, please fill in the following details and use the tag [bug].

**Describe the bug**
When multiplying a vector by a FactoredMatrix, an extra dimension is introduced into the returned value.

**Code example**
See tests in [tests/unit/factored_matrix/test_multiply_by_vector.py](https://github.com/neelnanda-io/TransformerLens/blob/a65657e1adb2894715ed7eae670e1a98fe2c97f0/tests/unit/factored_matrix/test_multiply_by_vector.py#L28-L67)

```python
@pytest.mark.skip(
    """
    This test fails with
    AssertionError: The values for attribute 'shape' do not match: torch.Size([1, 4]) != torch.Size([4]).
    I think this is a bug in __rmatmul__:
    `return ((other.unsqueeze(-2) @ self.A) @ self.B).squeeze(-1)` is squeezing on the wrong dimension.
    We could fix this by changing the argument to squeeze to `-2`. However there would be a small risk of
    breaking consumers that assume the existing behaviour?
    """
)
def test_right_matmul_by_vector():
    a = torch.rand(2, 3)
    b = torch.rand(3, 4)

    fm = FactoredMatrix(a, b)
    vector = torch.rand(2)

    assert_close(vector @ fm, vector @ (a @ b))
```


**System Info**
n/a

**Additional context**
n/a

### Checklist

- [ ] I have checked that there is no similar [issue](https://github.com/neelnanda-io/TransformerLens/issues) in the repo (**required**)

--- a/tests/unit/factored_matrix/test_multiply_by_vector.py
+++ b/tests/unit/factored_matrix/test_multiply_by_vector.py
@@ -25,18 +25,6 @@ def test_left_matmul_by_vector_leading_dim():
     assert_close(fm @ vector, (a @ b) @ vector)
 
 
-@pytest.mark.skip(
-    """
-    This test fails with
-    AssertionError: The values for attribute 'shape' do not match: torch.Size([1, 4]) != torch.Size([4]).
-
-    I think this is a bug in __rmatmul__:
-    `return ((other.unsqueeze(-2) @ self.A) @ self.B).squeeze(-1)` is squeezing on the wrong dimension.
-
-    We could fix this by changing the argument to squeeze to `-2`. However there would be a small risk of
-    breaking consumers that assume the existing behaviour?
-    """
-)
 def test_right_matmul_by_vector():
     a = torch.rand(2, 3)
     b = torch.rand(3, 4)
@@ -47,16 +35,6 @@ def test_right_matmul_by_vector():
     assert_close(vector @ fm, vector @ (a @ b))
 
 
-@pytest.mark.skip(
-    """
-    See comment for [test_right_matmul_by_vector].
-
-    This time, error is
-    AssertionError: The values for attribute 'shape' do not match: torch.Size([6, 1, 4]) != torch.Size([6, 4]).
-
-    Fix would be the same.
-    """
-)
 def test_right_matmul_by_vector_leading_dim():
     a = torch.rand(6, 2, 3)
     b = torch.rand(6, 3, 4)

--- a/transformer_lens/FactoredMatrix.py
+++ b/transformer_lens/FactoredMatrix.py
@@ -58,7 +58,7 @@ class FactoredMatrix:
             ), f"Left matrix must match on inner dimension, shapes were self: {self.shape}, other:{other.shape}"
             if other.ndim < 2:
                 # It's a vector, so we collapse the factorisation and just return a vector
-                return ((other.unsqueeze(-2) @ self.A) @ self.B).squeeze(-1)
+                return ((other.unsqueeze(-2) @ self.A) @ self.B).squeeze(-2)
             elif self.ldim > self.mdim:
                 return FactoredMatrix(other @ self.A, self.B)
             else:

